Documents/public-api.md

```diff
@@ -0,0 +1,34 @@
+# HMDA Platform Public API
+
+This documenatation describes de public HMDA Platform HTTP API
+
+## Institutions
+
+### Search
+
+* `/institutions?domain=<domain>`
+
+   * `GET` - Returns a list of institutions filtered by their email domain. If none are found, an HTTP 404 error code (not found) is returned
+
+   Example response, with HTTP code 200:
+
+   ```json
+   [
+      {
+        "id": "0",
+        "name": "Bank 0",
+        "domains": ["test@bank0.com"],
+        "externalIds":[
```

Comment:
`externalIds` are a little different in the current API.  

```scala
      "externalIds": [
        { "name": "RSSD ID", "value": "3333333" }, 
        { "name": "Federal Tax ID", "value": "33-3333333" }, 
        { "name": "NCUA Charter Number", "value": "333333" }
      ]

```

As you can see, there's a human-friendly `name` attribute, no `idType`, and `id` is called `value`.  I'm cool switching to `id` if you prefer that naming, and `idType` could be useful in the future too.  However, I do think adding `name` would be really helpful.  Without this, we'll have to map `idType` to the friendly name in the Keycloak frontend code.  If we add it to the API, it will be consistent across Keycloak and any other places we show this in the future.

    To do this, looks like we'd need to add a new attribute to [`ExternalId`](https://github.com/cfpb/hmda-platform/blob/379a6f021178c59636527388904630f7fc8b2f0f/model/shared/src/main/scala/hmda/model/institution/ExternalId.scala).

    I think the names could pretty closely follow the names of the case classes:

    ```scala
    case object FdicCertNo extends ExternalIdType("fdic-certificate-number", "FDIC Certificate Number")
    case object FederalTaxId extends ExternalIdType("federal-tax-id", "Federal Tax ID")
    case object NcuaCharterId extends ExternalIdType("ncua-charter-id", "NCUA Charter Number")
    case object OccCharterId extends ExternalIdType("occ-charter-id", "OCC Charter Number")
    case object RssdId extends ExternalIdType("rssd-id", "RSSD ID")
    case object UndeterminedExternalId extends ExternalIdType("undetermined-external-id", "???")
    ```

    I guess we also need to figure out what to do with `UndeterminedExternalId`?  That doesn't seem like something we'd want the end-user to see.  I'm not quite sure what an `UndeterminedExternalId` would even be.  Don't we always know where a given ID comes from?